lib/core_ext/numpy.rb

```diff
@@ -59,6 +73,44 @@ def roll(shift, axis: 0)
     end
   end
 
+  def assign(x1, y1, x2, y2, value)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] = value } }
+
+    self
+  end
+
+  def paste(x, y, other)
+    raise ArgumentError, 'must be in bounds' if x < 0 || y < 0 || x + other.first.length >= first.length || y + other.length >= length
+
+    other.each.with_index { |row, y2| row.each.with_index { |_val, x2| self[y2 + y][x2 + x] = other[y2][x2] } }
+
+    self
+  end
+
+  def adjust(x1, y1, x2, y2, difference)
+    pts!(x1, y1, x2, y2)
+
+    x1.upto(x2) { |x| y1.upto(y2) { |y| self[y][x] += difference } }
+
+    self
+  end
```

Comment:
this seems to specific, remove? Have you used this more than once? If you really want to keep it, let's change the name to `inc` or something

Reply:
This is the function I'm trying to replicate:

``` python
np.add(grid[x1:x2 + 1, y1:y2 + 1], diff)
```